sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathFunctionsSuite.scala

```diff
@@ -154,6 +155,19 @@ class MathFunctionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     testUnary(Floor, math.floor)
   }
 
+  test("factorial") {
+    val data = (0 to 20)
+    data.foreach { value =>
+      checkEvaluation(Factorial(Literal(value)), IntMath.factorial(value), EmptyRow)
+    }
+    checkEvaluation((Literal.create(null, IntegerType)), null, create_row(null))
+    val e = intercept[RuntimeException] {
+      checkEvaluation(Factorial(Literal(21)), null, EmptyRow)
+    }
+    assert(e.getCause.getMessage ==
```

Comment:
`===` instead of `==`, the former will give more information when it fails in the assert.